### PR TITLE
Add ability to love songs through listening services

### LINKED ISSUE #1424: Auto like on lastfm if a person likes a track on scrobbleable service
### Expected behavior
If you like a track, for example, on youtube - it should also like that track on lastfm

### Current behavior
Nothing happens.

### PR DESCRIPTION
**Pull request by yayuyokitano**
Closes #1424 
Really dont know about these labels in options, I think they can be improved a lot.
Adds properties to connector:
`Connector.isLoved`: function that returns true/false/nullish for whether a song is loved on website or not.
`Connector.loveButtonSelector`: selector or selectors for love/like button. When this is visible the song is interpreted as not loved.
`Connector.unloveButtonSelector`: selector or selectors for unlove/unlike button. When this is visible the song is interpreted as loved.

Only one of these three should be specified in any given connector similar to play/pause button selectors and isplaying function.

Property has been implemented to youtube and spotify connectors so far, but I think we could do with a few more. You can see how it works in practice though in those connector implementations.

### CODE REVIEW

File: src/core/content/connector.ts [hunk 1]
```diff
@@ -614,6 +643,29 @@ export default class BaseConnector {
...
+	public get isLovedCallback():
+		| ((isLoved: boolean | null) => Promise<void>)
+		| null {
+		return this._isLovedCallback;
+	}
+
+	public set isLovedCallback(
+		callback: (isLoved: boolean | null) => Promise<void>,
+	) {
+		callback(this.isLoved() ?? null);
```
Comment (inverse) [inline src/core/content/connector.ts:665]: is the `?? null` to ensure either `bool` or `null`?

File: src/core/object/controller/controller.ts [hunk 2]
```diff
@@ -567,6 +567,60 @@ export default class Controller {
...
+
+		/**
+		 * State did not change, don't do anything.
+		 */
+		if (this.currentSong.flags.isLovedInService === isLoved) {
+			return;
+		}
+
+		/**
+		 * song already had fetched a definitive loved state,
```
Comment (inverse) [inline src/core/object/controller/controller.ts:608]:
```suggestion
		 * Song already had fetched a definitive loved state,
```

**Review by inverse — APPROVED** [review]
Change seems nice - awesome work :+1: the callback stuff in src/core/content/connector.ts is a bit hard to follow  but it works :)

File: src/core/content/connector.ts (hunk 1 above)
Comment (yayuyokitano) [reply src/core/content/connector.ts:665]: Correct, I allow connectors to return undefined on no result to be more lenient on connector creators, in line with other connector functions.

**Comment by yayuyokitano** [comment]
> Change seems nice - awesome work 👍 the callback stuff in src/core/content/connector.ts is a bit hard to follow but it works :)

Yeah, I think maybe we want to rethink this structure now that controller and connector are tightly coupled anyway, but for now I have just maintained the same structure as our existing callback to be consistent.
